src/node_crypto_bio.h

```diff
@@ -96,6 +96,21 @@ class NodeBIO : public MemoryRetainer {
     return length_;
   }
 
+  // Provide a hint about the size of the next pending set of writes. TLS
+  // writes records of a maximum length of 16k of data plus a 5-byte header,
+  // a MAC (up to 20 bytes for SSLv3, TLS 1.0, TLS 1.1, and up to 32 bytes
+  // for TLS 1.2), and padding if a block cipher is used.  If there is a
+  // large write this will result in potentially many buffers being
+  // allocated and gc'ed which can cause long pauses.   By providing a
+  // guess about the amount of buffer space that will be needed in the
+  // next allocation this overhead is removed.
+  inline void set_allocate_tls_hint(size_t size) {
+    constexpr size_t kThreshold = 16 * 1024;
+    if (size >= kThreshold) {
+      allocate_hint_ = ((size / kThreshold + 1) * (kThreshold + 5 + 32));
```

Comment:
Superfluous parentheses:
```suggestion
      allocate_hint_ = (size / kThreshold + 1) * (kThreshold + 5 + 32);
```
The formula sets `allocate_hint_` to > 200% of `size` when the latter is at or near the threshold (e.g. `allocate_hint_=32842` when `size=16384`)

I assume that's intentional but the comment could make that clearer still.

Reply:
Cleaned up the parens...  

I chose to over allocate one TLS segment with framing overhead to try to make sure the entire write produced by SSL_write is in one allocation. OpenSSL may decide to write less than 16k in one TLS segment and that increases the framing overhead.   I can't predict all of the vagaries of TLS so that is why I'm calling this a hint not a firm calculation of what will be needed.